src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java

```diff
@@ -0,0 +1,166 @@
+        BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+        int count = 0;
+        try {
+            for (SourceObject object : SourceObject.values()) {
+                String name = object.getValue();
+                BrowseEntity.Builder entity = (BrowseEntity.builder(String.valueOf(object), String.valueOf(object),
+                                ENTITY_TYPE_OBJECTS).
+                        canBrowse(false).canSample(true));
+                browseDetailBuilder.addEntity(entity.build());
+                count++;
+            }
+        } catch (Exception e) {
+            throw new IOException("Unable to create a Connection", e);
+        }
+        return browseDetailBuilder.setTotalCount(count).build();
+    }
+
+    @Override
+    public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest)
+            throws IOException {
+        ConnectorSpec.Builder specBuilder = ConnectorSpec.builder();
+        Map<String, String> properties = new HashMap<>();
+        properties.put(ConfigUtil.NAME_USE_CONNECTION, "true");
+        properties.put(ConfigUtil.NAME_CONNECTION, connectorSpecRequest.getConnectionWithMacro());
+        String tableName = connectorSpecRequest.getPath();
+        if (tableName != null) {
+            properties.put(Constants.Reference.REFERENCE_NAME, ReferenceNames.cleanseReferenceName(tableName));
+            properties.put(MarketingCloudConstants.PROPERTY_OBJECT_NAME, SourceObject.valueOf(tableName).getValue());
+            schema = config.getSchema(SourceObject.valueOf(tableName));
+            specBuilder.setSchema(schema);
+        }
+        return specBuilder.addRelatedPlugin(new PluginSpec(MarketingCloudConstants.PLUGIN_NAME, BatchSource.PLUGIN_TYPE,
+                properties)).addRelatedPlugin(new PluginSpec(MarketingCloudDataExtensionSink.PLUGIN_TYPE,
+                BatchSink.PLUGIN_TYPE, properties)).build();
+    }
+
+    @Override
+    public List<StructuredRecord> sample(ConnectorContext connectorContext, SampleRequest sampleRequest)
+            throws IOException {
+        String object = sampleRequest.getPath();
+        if (object == null) {
+            throw new IllegalArgumentException("Path should contain object.");
+        }
+        try {
+            return listObjectDetails(SourceObject.valueOf(object));
+        } catch (Exception e) {
+            throw new IOException("unable to fetch records", e);
+        }
+    }
+
+    private List<StructuredRecord> listObjectDetails(SourceObject sourceObject) throws ETSdkException
+            , IOException {
+        List<StructuredRecord> sample = new ArrayList<>();
+        MarketingCloudClient marketingCloudClient = MarketingCloudClient.create
+                (config.getClientId(), config.getClientSecret(),
+                        config.getAuthEndpoint(), config.getSoapEndpoint());
+        if (sourceObject == SourceObject.DATA_EXTENSION) {
```

Comment:
Changed it to 'SourceObject.DATA_EXTENSION == sourceObject'

Reply:
After done this change. When user will select data extension object. It will not throw the blank list if we follow this approach 'SourceObject.DATA_EXTENSION == sourceObject'. Because source object is a list of object. If we will provide DataExtension at first preference. Compiler will get confused. This is what I have noticed.

Reply:
Yes. Dhiraj was right. That time I was facing issue with this approach 'SourceObject.DATA_EXTENSION == sourceObject'. After doing invalidate caches. It worked on the same day.
Changed it to 'SourceObject.DATA_EXTENSION == sourceObject'